### ISSUE TITLE
limit_disclosure: 'required' for BBS+

### ISSUE THREAD

**Issue by OR13** [decentralized-identity/waci-presentation-exchange#6]
We need to either pass a frame OR generate a frame from input descriptors.

**Comment by OR13** [comment]
ping @tplooker @csuwildcat

**Comment by csuwildcat** [comment]
I believe you can generate a framing object from what exists in PE using common utilities found in most JSONPath libs, which we can specify as a non-normative process. Would the following (rough strawman) processing instructions suffice:

1. Generate a new JSON object in memory.
2. Use the JSONPath selectors to detect matches on a target Holder object you are inspecting
3. If the selector finds a value, deep set that path in your empty JSON object with the framing path marker, using something like this: https://github.com/dchester/jsonpath#jpapplyobj-pathexpression-fn
4. Repeat this for each field, until your framing object paths are fully populated
5. Do whatever else you need to do to mint your framing object

What do you think?

**Comment by OR13** [comment]
This would appear to be required If verifier wants too convey the exact fields a BBS+ VC should disclose... which is then wrapped in the VP and submitted

**Comment by kdenhartog** [comment]
The frame will need to include more than just the property keys to be able to work correctly. In particular the context values will need to be added and the type values will need to be added and I'm not sure if the PE semantics have everything that we'd need to map it to a frame at this point. Is anyone familiar with how frames work and how PE work able to comment on how the semantics of the type and context URIs are conveyed at this point?

I really need to do a deep dive on the PE work more before I can be of more help.

**Comment by OR13** [comment]
@kdenhartog I think most of the frame can be pulled from the type IRI and field names, but yes.... I worry that edge cases will not be well covered.... in particular, building a nested data structure in the frame from PE seems hard.

**Comment by csuwildcat** [comment]
I agree with @OR13  that I won't feel 100% about this being easy to do in all cases, but it does look doable at first blush. I think Orie is exactly right to say that we need to fully understand and test the range of possible descriptors and frames to know if there are any issues with translating between them.

**Comment by OR13** [comment]
This issue is not critically blocking progress, either we provide evidence that its possible or we inject a JSON-LD frame for v0.1, since I know the frame solution is possible, I am in favor on simply doing what we know will work, and will align with existing tooling.

@csuwildcat @tplooker how hard will it be to add an optional `frame` parameter to PE Spec, that seems like the only path forward on this given the timeline.... we can't afford to "keep hoping things will work".

**Comment by OR13** [comment]
On the call, we discussed now we should provide the frame next the input descriptors.

We now require an example that matches the messages

**Comment by OR13** [comment]
Decision was made not to use the `limit_disclosure` feature of the PE spec and use a JSON-LD frame for now.

**Comment by OR13** [comment]
I opened https://github.com/decentralized-identity/presentation-exchange/issues/234 to track the requested change in the upstream spec.

**Comment by brentzundel** [comment]
link to AcaPy's implementation that derives a JSON-LD frame object from the `input_descriptors` object in a presentation definition: https://github.com/hyperledger/aries-cloudagent-python/blob/34c554440c6eb05c671ffa827b3a72075486e0ae/aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py#L504

**Comment by brentzundel** [comment]
Our side is done, leaving open to keep link alive to PE spec issue https://github.com/decentralized-identity/presentation-exchange/issues/234

**Comment by brianorwhatever** [comment]
so realizing now since we are saying we are using both the frame and input descriptors there is some point that either:
1. implementer converts from one to the other
2. user selects both the frame and input descriptors

obviously 2 is an awkward experience so I think people will most likely implement a conversion themselves.. I think this is fine for v0.1 but I think we might want to rethink this approach for v1
